Feed/serializer

alehakad opened the pull request.

alehakad committed:
Feed app created

alehakad committed:
Feed serializer added

alehakad committed:
Reaction added to serializer

alehakad committed:
depth removed

alehakad committed:
Merge branch 'develop' into feed/serializer

alehakad committed:
black applied

alehakad committed:
recipe preview image added

alehakad committed:
preview added to serializers.py

alehakad committed:
Feed serializer fixes

alehakad committed:
Feed serializer reactions fixed

alehakad committed:
Apiview changed to ListModelMixin

vBaMnup commented on src/apps/feed/apiviews.py:
```diff
@@ -0,0 +1,90 @@
+from django.db.models import Sum, Count, F
+
+
+class FeedSubscriptionsPagination(PageNumberPagination):
```
Используй для пагинаций отдельный файл pagination.py

vBaMnup commented on src/apps/feed/apiviews.py:
```diff
@@ -0,0 +1,90 @@
+
+
+class FeedSubscriptionsPagination(PageNumberPagination):
+    page_size = 5
```
Давай все цифры, которые у нас возможно будут меняться, вынесем в settings.py

vBaMnup commented on src/apps/feed/apiviews.py:
```diff
@@ -0,0 +1,90 @@
+    page_size = 5
+
+
+class FeedPopularPagination(PageNumberPagination):
```
Нам, в принципе, одного класса пагинатора хватит :)

alehakad replied:
Ок, согласен

vBaMnup commented on src/apps/feed/apiviews.py:
```diff
@@ -0,0 +1,90 @@
+    page_size = 5
+
+
+class FeedSubscriptionsList(mixins.ListModelMixin, viewsets.GenericViewSet):
```
Лента нового и популярного, доступны неавторизированным пользователям. Лента подписок - только авторизированному

vBaMnup commented on src/apps/feed/apiviews.py:
```diff
@@ -0,0 +1,90 @@
+        comments_count=Count("comments"),
+        views_count=Count("views"),
+        reactions_count=Count("reactions"),
+        activity_count=F("comments_count") + F("views_count") + F("reactions_count"),
```
Активности мы считаем не за всё время, а за месяц (изначально было за сутки, но пока сделаем за месяц) :)

vBaMnup commented on src/apps/feed/serializers.py:
```diff
@@ -0,0 +1,54 @@
+class FeedSerializer(serializers.ModelSerializer):
+    comments_count = serializers.IntegerField()
+    views_count = serializers.IntegerField()
+    reactions = serializers.SerializerMethodField()
```
Можно использовать ReactionSerializer прямо тут и не создавать отдельный метод

vBaMnup commented on src/apps/feed/serializers.py:
```diff
@@ -0,0 +1,54 @@
+    views_count = serializers.IntegerField()
+    reactions = serializers.SerializerMethodField()
+    tag = TagListSerializerField()
+    author = serializers.CharField(source="author.username")
```
сделай сериализатор для автора, и возвращай его id, username и аватар

vBaMnup commented on src/apps/feed/serializers.py:
```diff
@@ -0,0 +1,54 @@
+            "activity_count",
+        ]
+
+    # def get_comments_count(self, obj):
```
закомментированный код удали

vBaMnup commented on src/apps/reactions/models.py:
```diff
@@ -25,7 +25,9 @@ class Reaction(models.Model):
-        ContentType, limit_choices_to=limit_models, on_delete=models.CASCADE
+        ContentType,
+        limit_choices_to=limit_models,
+        on_delete=models.CASCADE,
```
set_null

vBaMnup commented on src/base/services.py:
```diff
@@ -16,6 +16,10 @@ def user_avatar_path(instance, filename):
 
 
+def recipe_preview_path(instance, filename):
+    return f"preview/recipe_{instance.slug}/{filename}"
```
id

alehakad replied:
Лучше slug по по-моему, так как при создании из админки id будет None

vBaMnup replied:
ну пусть будет слаг, хотя не очень красиво

alehakad committed:
Paginator file created, serializer fixed

alehakad committed:
activity_count filtered by last month

alehakad committed:
viewsets joined to one

vBaMnup committed:
Transferring the view logic from apiview.py to views.py;
Changing the logic of getting queryset feed;
RecipeView model: remove count and count increment method;
Cleaning up code and imports;
Added activity_interval to settings.py;

alehakad committed:
ViewRecipe count test removed

alehakad committed:
ViewRecipe count test removed

alehakad committed:
subscription filter fixed

alehakad committed:
tz awareness added to time

alehakad committed:
subscriptions filter fixed

alehakad committed:
total comments, views, reaction counts added
